Controller/Adminhtml/Account/Open.php

```diff
@@ -34,87 +34,95 @@
 use Nosto\Tagging\Model\User\Builder as NostoCurrentUserBuilder;
 
-/**
- * Iframe block for displaying the Nosto account management iframe.
- * This iframe is used to setup and manage your Nosto accounts on a store basis
- * in Magento.
- */
-class Iframe extends BlockTemplate
+class Open extends Base
 {
-    use IframeTrait;
-
-    public const IFRAME_VERSION = 1;
+    use ConnectionTrait;
 
-    private NostoHelperAccount $nostoHelperAccount;
     private Session $backendAuthSession;
-    private NostoIframeMetaBuilder $iframeMetaBuilder;
-    private NostoCurrentUserBuilder $currentUserBuilder;
+    private NostoHelperAccount $nostoHelperAccount;
     private NostoHelperScope $nostoHelperScope;
+    private NostoConnectionMetadataBuilder $connectionMetadataBuilder;
+    private NostoCurrentUserBuilder $currentUserBuilder;
     private NostoLogger $logger;
 
     /**
-     * Constructor.
-     *
-     * @param BlockContext $context the context.
-     * @param NostoHelperAccount $nostoHelperAccount the account helper.
+     * @param Context $context
      * @param Session $backendAuthSession
-     * @param NostoIframeMetaBuilder $iframeMetaBuilder
-     * @param NostoCurrentUserBuilder $currentUserBuilder
+     * @param NostoHelperAccount $nostoHelperAccount
      * @param NostoHelperScope $nostoHelperScope
+     * @param NostoConnectionMetadataBuilder $connectionMetadataBuilder
+     * @param NostoCurrentUserBuilder $currentUserBuilder
      * @param NostoLogger $logger
-     * @param array $data
      */
     public function __construct(
-        BlockContext $context,
-        NostoHelperAccount $nostoHelperAccount,
+        Context $context,
         Session $backendAuthSession,
-        NostoIframeMetaBuilder $iframeMetaBuilder,
-        NostoCurrentUserBuilder $currentUserBuilder,
+        NostoHelperAccount $nostoHelperAccount,
         NostoHelperScope $nostoHelperScope,
-        NostoLogger $logger,
-        array $data = []
+        NostoConnectionMetadataBuilder $connectionMetadataBuilder,
+        NostoCurrentUserBuilder $currentUserBuilder,
+        NostoLogger $logger
     ) {
-        parent::__construct($context, $data);
+        parent::__construct($context);
 
-        $this->nostoHelperAccount = $nostoHelperAccount;
         $this->backendAuthSession = $backendAuthSession;
-        $this->iframeMetaBuilder = $iframeMetaBuilder;
-        $this->currentUserBuilder = $currentUserBuilder;
+        $this->nostoHelperAccount = $nostoHelperAccount;
         $this->nostoHelperScope = $nostoHelperScope;
+        $this->connectionMetadataBuilder = $connectionMetadataBuilder;
+        $this->currentUserBuilder = $currentUserBuilder;
         $this->logger = $logger;
     }
 
     /**
-     * Gets the iframe url for the account settings page from Nosto.
+     * @return Redirect
+     * @suppress PhanUndeclaredMethod
+     * @noinspection PhpPossiblePolymorphicInvocationInspection
```

Comment:
I think those are invalid. Please check your IDE path mapping settings.

Reply:
Suppressions? I think these are correct. Without the `PhanUndeclaredMethod` it will complain about the `setUrl` function not existing in the interface. Without the `PhpPossiblePolymorphicInvocationInspection` it will complain about `Potentially polymorphic call`.

Reply:
Check your IDE, I think your path mappings are not including Magento's core:
![image](https://user-images.githubusercontent.com/2778820/182108360-3f06dc3a-ff02-403d-979b-25c1907d7fbc.png)